### ISSUE TITLE
Request: Publish binaries as NPM package

### ISSUE THREAD

**Issue by mattrossman** [tmedwards/tweego#19]
It would be super handy to be able to bootstrap a Tweego project with a simple

`npm install tweego`

and then be able to set up a `package.json` with scripts like
```json
"scripts": {
  "start": "tweego -w -o dist/index.html src/index.twee & live-server dist",
  "build" "tweego -o dist/index.html src/index.twee"
}
```
This would make projects more portable since the Tweego dependency would be clearly defined and automatically pulled along with any other web development tools like `live-server` on an `npm install`, rather than requiring users to manually download the appropriate binary and add it to their PATH.

**Comment by Uzume** [comment]
I am not sure this is a good idea. Tweego can be used in projects on systems that do not even use `node` and `npm` (much less have them installed even). I am aware binaries can be packaged with `npm` packages and some people are using it as a generic packaging tool but remember it was originally designed to be a package tool for JavaScript and although tweego manipulates text that typically includes some JavaScript, it does not in fact contain any itself (short of prepackaging some twine story formats that do use JavaScript).

There already are some other projects around that have tweego setups similar to what you are requesting, e.g.,
- [ChapelR/tweego-setup](https://github.com/ChapelR/tweego-setup)

**Comment by mattrossman** [comment]
To clarify, I'm not suggesting the current distribution channel be removed, just that the binaries be mirrored on npmjs. Nothing would change for folks that don't use NPM. That being said, given the audience that Tweego is targeting I think you'd be hard pressed to find a user that doesn't already have NPM installed.

The example you shared requires users to manually install Tweego on their system and set the $PATH, which adds friction. NPM would be a convenience so you only need to run a single command to have a working dev environment. Also it would ensure your dev tools are versioned together, so the builds run the same on each system.

Just for a simple project I find myself using 3 other NPM tools (`live-server` for a dev server, `cpx` for watching static files like images and audio, and `concurrently` for running everything together), so Tweego is the awkward black sheep. When deploying my story on Netlify, I have to write a build script to install Tweego and modify the $PATH whereas npm dependencies get installed by default.

To me, it's not important what language the tool was written in, it's about the use case. Tweego is a tool for building web pages, and NPM is the defacto package manager for web developers.

Related: https://www.npmjs.com/package/go-npm

**Comment by Uzume** [comment]
@mattrossman Perhaps you have valid points, however:
> [...]I think you'd be hard pressed to find a user that doesn't already have NPM installed.

I think I qualify here as I use tweego often and rarely on systems where I am using `npm`. Perhaps I am not your average user though as I use it most often to decompile Twine stories to twee code with `-d` (I enjoy many engagingly well written stories that unfortunately are also often poorly written from a technical perspective and decompiling helps when no or only highly dated walkthroughs are available; without such it is very easy to miss/bypass large portions of the work).

> The example you shared requires users to manually install Tweego on their system and set the $PATH, which adds friction.

I agree that example is dated and could be better but the point I was trying to make is nothing is keeping you from repackaging tweego and distributing it yourself on `npmjs` as you suggest (not unlike how @ChapelR created his own setup framework).

I appreciate the clarification.

**Comment by mattrossman** [comment]
Good idea, I've never published on npm before but I'll look into how I might repackage it :)

**Comment by tmedwards** [comment]
I've been resistant to this idea in the past for various reasons, but I suppose an NPM package wouldn't be the end of the world.

I'll look into it.

**Comment by mattrossman** [comment]
I've put together a working example here:

https://github.com/mattrossman/tweego-node

Right now it just downloads version 2.1.1 but ideally the node package versions would mirror the Tweego version.

**Comment by Uzume** [comment]
You will likely want to look at something like: https://api.github.com/repos/tmedwards/tweego/releases
